bool complete_task
---
bool update_aog